### PR TITLE
Improve the telemetry device "Segment Replication Stats" by changing to parse JSON format and using numeric bytes value

### PR DESCRIPTION
**Pull request by tlfeng**
### Description

Although PR https://github.com/opensearch-project/opensearch-benchmark/issues/339 added new telemetry device "Segment Replication Stats", there are some problems. The PR will solve the problem.
- Fix the problem of using single whitespace `str.split(" ")` as the delimiter to parse the REST API response, where maybe more than 1 space between adjacent values.
- Change to parse the REST API response in JSON format to improve the robustness.
- Change the byte value field to be purely numeric to be directly comparable.

For detail, see https://github.com/opensearch-project/opensearch-benchmark/issues/339#issuecomment-1672297062.

### Issues Resolved
Address a problem related to issue https://github.com/opensearch-project/opensearch-benchmark/issues/339

The screenshot illustrates the current problem. Sometime when there are more than 1 space between adjacent values, it leads mismatch in the fields and values of the result.
Problem with parsing the API response

### Testing
- [x] New functionality includes testing

End to end test:
Verified that every field has correct value.

Command used: `opensearch-benchmark execute-test --workload=http_logs --pipeline=benchmark-only \
--target-hosts=opens-clust-xxx.elb.us-west-2.amazonaws.com:80 \
--include-tasks=delete-index,create-index,check-cluster-health,index-append \
--workload-params='{"index_settings":{"number_of_shards": 1, "number_of_replicas": 1 }}' \
--telemetry=segment-replication-stats --test-mode `

image

image

image

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by tlfeng** [comment]
@mch2 @IanHoang @gkamat Looking forward to your opinions. Thank you! I'd like to make this change into the next release of version 1.1.0 .

**Comment by tlfeng** [comment]
End to end test:
Verified that every field has correct value.
image

image

image

File: tests/telemetry_test.py [hunk 1]
```diff
@@ -3431,27 +3449,27 @@ def test_stores_default_stats(self, metrics_store_put_doc):
 
         metrics_store_put_doc.assert_has_calls([call({
             "name": "segment-replication-stats",
-            "shard_id": "[so][0]",
-            "target_node": "node-1",
-            "target_host": "127.0.0.1",
+            "shard_id": "[logs-211998][6]",
```
Comment (dreamer-89) [inline tests/telemetry_test.py:3452]: nit: Is `shard_id` change needed here ?

**Review by dreamer-89 — APPROVED** [review]

File: tests/telemetry_test.py (hunk 1 above)
Comment (tlfeng) [reply tests/telemetry_test.py:3452]: @dreamer-89 Thanks a lot for your review! 👏 The change is indeed not necessary, I just updated the test with a real API response from my recent performance test. I think it's fine to keep the changes.

**Comment by tlfeng** [comment]
Manually tested again after the last commit
image
image

**Review by IanHoang — APPROVED** [review]
LGTM. Thanks for discovering this bug and providing the fix @tlfeng!

**Pull request merged**